src/System.Private.CoreLib/shared/System/DateTime.cs

```diff
@@ -641,32 +641,38 @@ public int CompareTo(DateTime value)
 
         // Returns the tick count corresponding to the given year, month, and day.
         // Will check the if the parameters are valid.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static long DateToTicks(int year, int month, int day)
         {
-            if (year >= 1 && year <= 9999 && month >= 1 && month <= 12)
+            if (year < 1 || year > 9999 || month < 1 || month > 12 || day < 1)
```

Comment:
```suggestion
            if ((uint)(year - 1) > (9999 - 1) || ((uint)(month - 1) > (12 - 1) || day < 1)
```
Although `year - 1` is used again in L658, so this could be combined.
Same for `month - 1`, `day - 1` (they are used later in the method).

Reply:
thanks for the suggestion but I think that it would rather not increase the perf but decrease the readability of the source code